backend/composer/services/cs_ingestion/cs_ingestion_services.py

```diff
@@ -40,253 +39,351 @@
+
+    if len(statement[SENTENCE_NUMBER]) > 1:
+        logging.warning(f'Multiple sentence numbers found in statement for {statement[LABEL]}. Using the first one.')
+
     sentence, created = Sentence.objects.get_or_create(
         doi__iexact=statement[ID],
-        defaults={"title": text[0:185], "text": text,
+        defaults={"title": text[0:185],
+                  "text": text,
                   "doi": statement[ID],
                   "external_ref": statement[SENTENCE_NUMBER][0] if has_sentence_reference else None,
-                  "batch_name": f"neurondm-{NOW}" if has_sentence_reference else None
+                  "batch_name": f"neurondm-{NOW}" if has_sentence_reference else None,
+                  "state": SentenceState.COMPOSE_NOW
                   },
     )
     if created:
         logging.info(f"Sentence for neuron {statement[LABEL]} created.")
         sentence.save()
-    return sentence
+    return sentence, created
 
 
-def has_prop(prop):
-    return True if len(prop) > 0 else False
+def should_overwrite(connectivity_statement: ConnectivityStatement, statement: Dict) -> bool:
+    if connectivity_statement.destinations.count() > 1:
+        logging.warning(f'Skip statement {statement[LABEL]} due to:'
+                        f' statement already found and has multiple destinations')
+        return False
 
+    if connectivity_statement.state != CSState.EXPORTED:
+        logging.warning(f'Skip statement {statement[LABEL]} due to:'
+                        f' statement already found and is not in {CSState.EXPORTED} state')
+        return False
 
-def get_value_or_none(model, prop):
-    if has_prop(prop):
-        if model == AnatomicalEntity:
-            return model.objects.filter(ontology_uri=prop[0])[0]
-        else:
-            return model.objects.get(ontology_uri=prop[0])
+    return True
+
+
+def get_or_create_connectivity_statement(statement: Dict, sentence: Sentence) -> Tuple[ConnectivityStatement, bool]:
+    reference_uri = statement[ID]
+    defaults = generate_connectivity_statement_defaults(statement, sentence)
+    connectivity_statement, created = ConnectivityStatement.objects.get_or_create(
+        reference_uri__exact=reference_uri,
+        defaults=defaults
+    )
+    if created:
+        update_many_to_many_fields(connectivity_statement, statement)
+
+    return connectivity_statement, created
+
+
+def generate_connectivity_statement_defaults(statement: Dict, sentence: Sentence) -> Dict:
+    return {
+        "sentence": sentence,
+        "knowledge_statement": statement[LABEL],
+        "sex": get_sex(statement),
+        "circuit_type": get_circuit_type(statement),
+        "functional_circuit_role": get_functional_circuit_role(statement),
+        "phenotype": get_phenotype(statement),
+        "projection_phenotype": get_projection_phenotype(statement),
+        "state": CSState.EXPORTED,  # TODO: Confirm if this is fine
```

Comment:
Previously the state change was running transitions instead of starting the state in the desired place already.
Because the previous logic needed updated I opted for this simple approach but I'm not sure if I'm missing some context

Reply:
@afonsobspinto see my previous comment about the `.update` method